### ISSUE TITLE
RHSM: entitlement cert changes while composer is running are never picked up

### ISSUE THREAD

**Issue by thozza** [osbuild/osbuild-composer#2175]
**Describe the bug**
The RHSM secrets available on the system are loaded only when osbuild-composer starts and never updated while it is running. 
https://github.com/osbuild/osbuild-composer/blob/b6acfed2041c56f707dc5dc456329fba421cafce/cmd/osbuild-composer/composer.go#L64
https://github.com/osbuild/osbuild-composer/blob/b6acfed2041c56f707dc5dc456329fba421cafce/internal/rpmmd/repository.go#L380

This creates a problem on for long running osbuild-composer instances, especially when the system is unregistered and registered again using subscription-manager, while osbuild-composer is running. In such case, the previously loaded RHSM certificates / keys (e.g. `/etc/pki/entitlement/<keyid>-key.pem`) may not be present on the disk any more and were replaced by a different keys and `redhat.repo` got updated.

In such situation, any calls to `dnf-json` with repos that use RHSM secrets will fail with errors such as:
```
 Errors during downloading metadata for repository '0':
   - Curl error (58): Problem with the local SSL certificate for https://cdn.redhat.com/content/dist/rhel8/8.3/x86_64/baseos/os/repodata/repomd.xml [could not load PEM client certificate, OpenSSL error error:02001002:system library:fopen:No such file or directory, (no key found, wrong pass phrase, or wrong file format?)]
```

One has to restart `osbuild-composer.service` to make it load RHSM secrets from the disk again to fix the issue.

**Environment**
 * OS version (`/etc/os-release` and `/etc/redhat-release`): RHEL
 * osbuild-composer version (`rpm -qi osbuild-composer)` current, but reproduced with v33.2 (RHEL-8.5)

**To Reproduce**

1. install osbuild-composer on RHEL system
2. subscribe the RHEL system using subscription-manager
3. start osbuild-composer e.g. by interacting with it using `composer-cli`
4. create a blueprint in osbuild-composer
5. unregister the system using subscription-mnaager
6. register the system using subscription-manger again
7. start a new compose in osbuild-composer

**Expected behavior**
osbuild-composer should correctly pick the most recent entitlement certificates and pass them to `dnf-json` when depsolving packages for an image.

**Additional context**
The entitlement certificates may change (be regenerated) even without re-registering the system, e.g. when a customer account switches to use SCA, etc.

**Comment by thozza** [comment]
RHSM provides internal DBus API, which could be used, but may change at some point. This was suggested by RHSM devel.
https://www.candlepinproject.org/docs/subscription-manager/dbus_objects.html#signals-2

**Comment by thozza** [comment]
Fixed by #2751
